### PR TITLE
Canonical Tags Improvements.

### PR DESCRIPTION
**Pull request by EugeneKyale**
Ticket : [TEC-4693](https://theeventscalendar.atlassian.net/browse/TEC-4693)

Contains the following updates.

-  Introduced a new filter i.e. `tribe_events_add_canonical_tag` to make the addition of canonical tags filterable.
- Added integrations for Yoast SEO and RankMath to prevent duplicate canonical tags.
- Added trailing slash to the canonical url.
- Added an ID i.e. `tec-seo-meta-tag` to the canonical tag to make it clear it was added by TEC.  

**Screenshot 📸**

How the canonical tag appears when none of the SEO plugins mentioned above are active.

image

[TEC-4693]: https://theeventscalendar.atlassian.net/browse/TEC-4693?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Camwyn — APPROVED** [review]

**Review by bordoni — DISMISSED** [review]
New integrations should use the new system in `TEC/Events/Integrations` we already have a YoastSEO integration.

**Comment by EugeneKyale** [comment]
> New integrations should use the new system in `TEC/Events/Integrartions` we already have a YoastSEO integration.

@bordoni I've made the updates to use the new integration system.

File: src/Events/Integrations/Plugins/Rank_Math/Provider.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	 */
+	public function load_conditionals(): bool {
+		$should_load = true;
+
+		if ( ! defined( 'RANK_MATH_FILE' ) || empty( RANK_MATH_FILE ) ) {
+			$should_load = false;
+		}
+
+		return $should_load;
+	}
```
Comment (Camwyn) [inline src/Events/Integrations/Plugins/Rank_Math/Provider.php:38]: Might be simpler:
```suggestion
	public function load_conditionals(): bool {
		return defined( 'RANK_MATH_FILE' ) && ! empty( RANK_MATH_FILE );
	}
```

**Review by Camwyn — CHANGES_REQUESTED** [review]
One suggestion

File: src/Events/Integrations/Plugins/Rank_Math/Provider.php (hunk 1 above)
Comment (EugeneKyale) [reply src/Events/Integrations/Plugins/Rank_Math/Provider.php]: @Camwyn thanks, this is better.

**Review by Camwyn — APPROVED** [review]

**Pull request merged**